From eddc6a58f84d4b89599101ff374290fba7e7645c Mon Sep 17 00:00:00 2001
From: Pablo Saavedra <psaavedra@igalia.com>
Date: Tue, 6 Sep 2022 11:24:07 +0200
Subject: [PATCH] cmake: Fix CMake (v3.16) cannot create ALIAS target
 "Wpe::FDO" error

... because target "PkgConfig::WpeFDO" is imported but not globally
visible in cmake/FindWpeFDO.cmake.

Notice that this issue is not reproducible in CMake v3.22.

CMake doc [1][2] says:

"The scope of the definition of an IMPORTED target is the directory
where it was defined. It may be accessed and used from
subdirectories, but not from parent directories or sibling directories.
The scope is similar to the scope of a cmake variable.

It is also possible to define a GLOBAL IMPORTED target which is
accessible globally in the buildsystem."

[1] 3.16: https://cmake.org/cmake/help/v3.16/manual/cmake-buildsystem.7.html#imported-targets
[2] latest: https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#imported-targets
---
 cmake/FindWpeFDO.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Upstream-Status: Submitted [https://github.com/Igalia/cog/pull/487]

diff --git a/cmake/FindWpeFDO.cmake b/cmake/FindWpeFDO.cmake
index 3a45757..b40a32e 100644
--- a/cmake/FindWpeFDO.cmake
+++ b/cmake/FindWpeFDO.cmake
@@ -1,5 +1,5 @@
 find_package (PkgConfig REQUIRED QUIET)
-pkg_check_modules(WpeFDO QUIET wpebackend-fdo-1.0>=1.8.0 IMPORTED_TARGET)
+pkg_check_modules(WpeFDO QUIET wpebackend-fdo-1.0>=1.8.0 GLOBAL IMPORTED_TARGET)
 
 if (TARGET PkgConfig::WpeFDO)
   add_library(Wpe::FDO ALIAS PkgConfig::WpeFDO)
-- 
2.34.1

